packages/yew/src/html/listener/mod.rs

```diff
@@ -37,51 +33,73 @@ pub enum ChangeData {
     /// to collect your required data such as `value` and `selected_index`.
     /// You can also iterate throught `selected_options` yourself, this does require adding the
     /// [web-sys](https://crates.io/crates/web-sys) crate with the `HtmlCollection` feature.
-    Select(SelectElement),
+    Select(HtmlSelectElement),
     /// Files
-    Files(FileList),
+    Files(web_sys::FileList),
 }
 
-fn oninput_handler(this: &Element, event: InputEvent) -> InputData {
+/// Extract target as Element from event
+fn extract_target(event: &Event) -> Element {
+    event
+        .target()
+        .expect("no target on event")
+        .dyn_into()
+        .unwrap()
+}
+
+pub(crate) fn cast_event<T>(e: Event) -> T
+where
+    T: wasm_bindgen::JsCast,
+{
+    e.unchecked_into()
+}
```

Comment:
This function seems rather unnecessary considering it's a one liner and `unchecked_into` already has this exact signature.

Reply:
It's there mostly to provide hints for rustc's type inference in the macro and avoid some verbosity by not having to type the signature over and over. I'll add it as a comment.